### ISSUE TITLE
Relative path concatenated wrong on Windows

### ISSUE THREAD

**Issue by hho** [asciidoctor/asciidoctor-maven-plugin#351]
For a different issue (?), I made a minimal reproducer repository: https://github.com/hho/asciidoctor-demo

@abelsromero had a look at it and it turned out, this doesn't even build on Windows. The PDF generation fails with `asciidoctor-diagram: ERROR: Failed to generate image: Unknown error (SystemCallError) - Unknown Error (20047) - C:\Users\asalgadr\temp\asciidoctor-demo\target\C:`

He asked me to open this issue. It seems, there are paths concatenated the wrong way.

**Comment by abelsromero** [comment]
Notes of what's is found so far:
- Rendering the same documents on Ruby with asciidoctor-diagram 1.5.9 works fine:
- Rendering the same documents on Maven with asciidoctor-diagram 1.5.8 & 1.5.9 (using gem-maven-plugin) fails (https://github.com/abelsromero/asciidoctor-demo/blob/analysis/pom.xml)
- Issue also reproduces for embedded diagrams.

Both in Windows and Linux the problem seems to be that asciidoctor-diagram is appending the absolute path of the generated image to the realitve path. In *nix this is translated into file not found, in Windows, it is an invalid path. However, since it does not reproduce for Ruby, I'd say it has something to do with maven or AsciidoctorJ.

**Comment by mojavelinux** [comment]
This could be related to https://github.com/asciidoctor/asciidoctor/issues/2869. There was a bunch of PathResolver fixes in 1.5.8 that *may* address this.

**Comment by abelsromero** [comment]
Still happens only when rendering PDF with AsciidoctorJ 1.5.8.1 and Asciidoctorj-diagram 1.5.11

**Comment by abelsromero** [comment]
As mentioned in the linked issue, the problem is not simple and seems to be located in asciidoctor itself.
As workarround we can set both attributes `imagesoutdir` and `diagram-cachedir` as in the example below:

```
  <imagesoutdir>${project.build.directory}/.asciidoctor-diagram</imagesoutdir>
  <diagram-cachedir>${project.build.directory}/.asciidoctor-diagram</diagram-cachedir>
```

It is possible to use `${java.io.tmpdir}` if we don't want files in target, but note that using clean will of course not delete them.

**Comment by abelsromero** [comment]
Confirmed there was an issue in Asciidoctor, this is fixed in the main branch and should be released with 2.0.11.
Details: https://github.com/asciidoctor/asciidoctor/issues/3778

**Comment by abelsromero** [comment]
Reproducer works fine with latest versions

```xml
<asciidoctorj.version>2.5.1</asciidoctorj.version>
<asciidoctorj-pdf.version>1.6.0</asciidoctorj-pdf.version>
<asciidoctorj-diagram.version>2.1.2</asciidoctorj-diagram.version>
```
